File: src/test/programmers/java/level2/SkillTree.java
package test.programmers.java.level2;

public class SkillTree {
    public static void main(String[] args) {
        String str = "CBD";
        String[] arr = {"BACDE", "CBADF", "AECB", "BDA"};

        System.out.println(solution(str, arr));
    }
    static public int solution(String skill, String[] skill_trees) {
        int answer = 0;

        for(String skillTree : skill_trees) {
            String temp = skillTree;

            for(int i = 0; i < skillTree.length(); i++) {
                String s = String.valueOf(skillTree.charAt(i));

                if(!skill.contains(s)) {
                    temp = temp.replace(s, "");
                }
            }

            System.out.println(temp);

            if(skill.indexOf(temp) == 0) {
                answer++;
            }
        }
        return answer;
    }
}

File: src/test/programmers/python/level2/SkillTree.py
def solution(skill, skill_trees):
    answer = 0
    for skill_tree in skill_trees:
        isOrdered = True
        for i in range(1, len(skill)):
            pre_idx = str(skill_tree).find(skill[i - 1])
            now_idx = str(skill_tree).find(skill[i])
            if now_idx != -1 and pre_idx == -1:
                isOrdered = False
                break
            if now_idx != -1 and pre_idx != -1:
                if now_idx <= pre_idx:
                    isOrdered = False
                    break
        if isOrdered:
            answer += 1

    return answer

print(solution("CBD", ["BACDE", "CBADF", "AECB", "BDA"]))